source/common/router/config_impl.cc

```diff
@@ -1816,11 +1816,18 @@ RouteSpecificFilterConfigConstSharedPtr PerFilterConfigs::createRouteSpecificFil
     Server::Configuration::ServerFactoryContext& factory_context,
     ProtobufMessage::ValidationVisitor& validator) {
   bool is_optional = (optional_http_filters.find(name) != optional_http_filters.end());
-  auto factory = Envoy::Config::Utility::getAndCheckFactoryByName<
-      Server::Configuration::NamedHttpFilterConfigFactory>(name, is_optional);
+  auto factory =
+      Envoy::Config::Utility::getFactoryByType<Server::Configuration::NamedHttpFilterConfigFactory>(
+          typed_config);
```

Comment:
Could we just use `Envoy::Config::Utility::getAndCheckFactory`?

Reply:
Sadly, nope. The `Envoy::Config::Utility::getAndCheckFactory` method requires that the proto message has `name()` and `type_config()` methods.

Reply:
Oh, yes, you are right. I have checked the code and we cannot use it.